lib/archive/tar/minitar/writer.rb

```diff
@@ -211,12 +211,15 @@ def add_file(name, opts = {}, &block) # :yields WriteOnlyStream, +opts+:
         raise Archive::Tar::Minitar::NonSeekableStream
       end
 
+      _, _, needs_long_name = split_name(name)
```

Comment:
This feels wrong to me that we’re calling `#split_name` twice (it's also being called in `#write_header`. I’m not quite sure what the API should be, and it would require changing `#add_file_simple` as well as `#add_file` as you’ve done here.

Maybe `#write_header` changes from `def write_header(long_name, header)` to `def write_header(header, short_name, prefix, needs_long_name)`. The call sites would change to `write_header(header, *split_name(name))` (line 167 in `#add_file_simple`) and your change would be to _not_ discard the first two return values and then pass them to `#write_header` in `#add_file`.

Reply:
Hi, I currently feel like I have to apologize for that because of the fact, that I tried to fix my actual problem with the least possible effort by not taking care of leaving the code base in a better state then before, or at least (in this case) not making it worse. I'm also a bit embarrassed because I should know better. :grimacing:
I'll start over, taking your comment into account.

Reply:
No apology necessary; I think that the minimal change you made it easy to understand (which is great as I don’t often visit this codebase). What I have suggested is a useful refactoring.

Reply:
I see. And thanks.